SU2_CFD/src/solvers/CGradientSmoothingSolver.cpp

```diff
@@ -0,0 +1,1011 @@
+      element_container[GRAD_TERM][EL_TRIA2] = new CTRIA3();
+    }
+  }
+
+  Residual = new su2double[nDim];   for (iDim = 0; iDim < nDim; iDim++) Residual[iDim] = 0.0;
+  Solution = new su2double[nDim];   for (iDim = 0; iDim < nDim; iDim++) Solution[iDim] = 0.0;
+
+  /*--- initializations for linear equation systems ---*/
+  if ( !config->GetSmoothOnSurface() ) {
+    nVar = config->GetSmoothSepDim() ? 1 : nDim;
+    LinSysSol.Initialize(nPoint, nPointDomain, nVar, 0.0);
+    LinSysRes.Initialize(nPoint, nPointDomain, nVar, 0.0);
+    Jacobian.Initialize(nPoint, nPointDomain, nVar, nVar, false, geometry, config, false, true);
+  } else {
+    if (config->GetSobMode() == ENUM_SOBOLEV_MODUS::PARAM_LEVEL_COMPLETE) {
+      Jacobian.Initialize(nPoint, nPointDomain, nDim, nDim, false, geometry, config, false , true);
+    } else {
+      LinSysSol.Initialize(nPoint, nPointDomain, 1, 0.0);
+      LinSysRes.Initialize(nPoint, nPointDomain, 1, 0.0);
+      Jacobian.Initialize(nPoint, nPointDomain, 1, 1, false, geometry, config, false, true);
+    }
+    visited.resize(geometry->GetnPoint(), false);
+  }
+
+  /*--- vectors needed for projection when working on the complete system ---*/
+  if (config->GetSobMode() == ENUM_SOBOLEV_MODUS::PARAM_LEVEL_COMPLETE ||
+      config->GetSobMode() == ENUM_SOBOLEV_MODUS::ONLY_GRAD || config->GetSobMode() == ENUM_SOBOLEV_MODUS::MESH_LEVEL) {
+    activeCoord.Initialize(nPoint, nPointDomain, nDim, 0.0);
+    helperVecIn.Initialize(nPoint, nPointDomain, nDim, 0.0);
+    helperVecOut.Initialize(nPoint, nPointDomain, nDim, 0.0);
+    helperVecAux.Initialize(nPoint, nPointDomain, nDim, 0.0);
+  }
+
+  /*--- Initialize the CVariable structure holding solution data ---*/
+  nodes = new CSobolevSmoothingVariable(nPoint, nDim, config);
+  SetBaseClassPointerToNodes();
+
+  /*--- Initialize the boundary of the boundary ---*/
+  if (config->GetSmoothOnSurface()) {
+
+    /*--- check which points are in more than one physical boundary ---*/
+    for (iPoint = 0; iPoint < nPoint; iPoint++) {
+      for (unsigned int iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+        if (config->GetMarker_All_KindBC(iMarker) != SEND_RECEIVE) {
+          long iVertex = geometry->nodes->GetVertex(iPoint, iMarker);
+          if (iVertex >= 0) {
+            marker_count++;
+          }
+        }
+      }
+      if (marker_count>=2) {
+        nodes->MarkAsBoundaryPoint(iPoint);
+      }
+      marker_count = 0;
+    }
+  }
+  nodes->AllocateBoundaryVariables();
+
+  /*--- vector for the parameter gradient ---*/
+  deltaP.resize(config->GetnDV_Total(), 0.0);
+
+  /*--- set the solver name for output purposes ---*/
+  SolverName = "SOBOLEV";
+}
+
+CGradientSmoothingSolver::~CGradientSmoothingSolver(void) {
+
+  if (element_container != nullptr) {
+    for (unsigned int iVar = 0; iVar < MAX_TERMS; iVar++) {
+      for (unsigned int jVar = 0; jVar < MAX_FE_KINDS; jVar++) {
+        delete element_container[iVar][jVar];
+      }
```

Comment:
isn't the base doing the deleting?

Reply:
Yes, this is already deleted in the base class.